moduledeps/module_test.go

```diff
@@ -191,13 +192,10 @@ func TestModulePluginRequirements(t *testing.T) {
 	m := &Module{
 		Name: "root",
 		Providers: Providers{
-			"foo": ProviderDependency{
+			addrs.NewLegacyProvider("foo"): ProviderDependency{
 				Constraints: discovery.ConstraintStr(">=1.0.0").MustParse(),
 			},
-			"foo.bar": ProviderDependency{
```

Comment:
So removed the `foo.bar` case? is that ... because this is invalid now? Or? Or can this case be added back/not removed?

Reply:
Good question! I addressed this in the PR comment and I know you've since seen it, but I'll reiterate for other reviewers: a side effect of this change is that the provider name, which could include alias, gets 'flattened' into just the provider type sooner than it used to.